renderdoc/driver/shaders/dxbc/dxbc_debug.cpp

```diff
@@ -573,6 +585,25 @@ bool State::Finished() const
   return dxbc && (done || nextInstruction >= (int)dxbc->GetNumInstructions());
 }
 
+void State::AssignValue(ShaderVariable &dst, uint32_t dstIndex, const ShaderVariable &src,
+                        uint32_t srcIndex)
+{
+  if(src.type == eVar_Float)
+  {
+    float ft = src.value.fv[srcIndex];
+    if(isinf(ft) || isnan(ft))
+      flags |= eShaderDebugStateFlags_GeneratedNanOrInf;
+  }
+  else if(src.type == eVar_Double)
+  {
+    double dt = src.value.dv[srcIndex];
+    if(isinf(dt) || isnan(dt))
+      flags |= eShaderDebugStateFlags_GeneratedNanOrInf;
+  }
```

Comment:
Do we have any tables that indicate opcode type? E.g. at its most basic a table of NUM_OPCODES size indicating whether an operation is arithmetic in nature, on up to something more like a description or flags structure for each opcode containing a variety of metadata along those lines. Closest I see is ExtendedOpcodeType but that isn't really what I'm looking for.

Checking that the type is float/double is a sort of proxy for that currently I suppose, as our expectation is that mask values will be int/uint. Attempting to do this more fine grained per your suggestion about comparing inputs vs outputs will probably end up with the opcode dispatch table getting fairly bloated, which we can do if you prefer, but I'd be more curious to see how things turn out in practice when using this. Similarly with a second "exception", it really depends on how broke your setup is... when we're debugging inf/nan it's usually a few scenarios:

1) a texture is aliased in some problematic way , and sampling of that value generates a nan/inf which is then used as an input to a lighting function or post process chain which propagates the values, often across multiple frames since we tend to do this in post chains with temporal history; could be multiple non-normal terms depending on history setup or aliasing complexity
2) a fixed function blend unit has generated a non-normal value into a render target which is later sourced; usually just a single non-normal
3) bad inputs to our lighting maths; this could be multiple terms